### PR TITLE
Improve Naming of internal Cache related classes

### PR DESCRIPTION
**Pull request by tresat**
Supercedes #22984.

It will definitely be easier to review this commit by commit.  The changes go from the top to the bottom of the table in [the Specification](https://docs.google.com/document/d/19s7NR8PxaBGD9HpyYXDHi4bjGa1fJjEyu97vdsSgfak/edit#), grouping changes to related types.  Each commit should be sane.

All changes here will be documented in the Specification.  This will be a very large change in terms of number of files, but should be very light in terms of complexity.

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
Finalized and should be in sync with the spec now.

File: subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/DefaultExecutionHistoryCacheAccess.java [hunk 1]
```diff
@@ -27,9 +27,9 @@
 public class DefaultExecutionHistoryCacheAccess implements ExecutionHistoryCacheAccess, Closeable {
     private final PersistentCache cache;
 
-    public DefaultExecutionHistoryCacheAccess(ScopedCache cacheRepository) {
-        this.cache = cacheRepository
-            .cache("executionHistory")
+    public DefaultExecutionHistoryCacheAccess(ScopedCacheBuilderFactory cacheBuilderFactor) {
```
Comment (ghale) [inline subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/DefaultExecutionHistoryCacheAccess.java:30]: I think this should be `cacheBuilderFactory`

**Review by ghale — APPROVED** [review]
Found one typo, but otherwise, this looks good to me.

File: subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/DefaultExecutionHistoryCacheAccess.java (hunk 1 above)
Comment (tresat) [reply subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/DefaultExecutionHistoryCacheAccess.java:30]: I'm seriously impressed you noticed this, looking at so many similar small changes!

**Pull request merged**